### ISSUE TITLE
-ldl missing for nmxt-plugin-scan

### ISSUE THREAD

**Issue by suedwestlicht** [Stazed/non-mixer-xt#42]
After some updates on my machine I had troubles building nmxt-plugin-scan, so the complete build failed:

```
[ 78%] Built target non-mixer-xt
[ 86%] Built target midi-mapper-xt
[ 87%] Linking CXX executable nmxt-plugin-scan
/usr/bin/ld: CMakeFiles/nmxt-plugin-scan.dir/src/ladspa/LADSPAInfo.C.o: undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [mixer/CMakeFiles/nmxt-plugin-scan.dir/build.make:313: mixer/nmxt-plugin-scan] Error 1
make[1]: *** [CMakeFiles/Makefile2:235: mixer/CMakeFiles/nmxt-plugin-scan.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

I had to add -ldl:

```
cd mixer
/usr/bin/c++ -std=c++17 -Wno-unused-parameter -Wno-unused-result -O3 -fomit-frame-pointer -ffast-math -pipe -msse -msse2 -mfpmath=sse -march=native -mtune=native "CMakeFiles/nmxt-plugin-scan.dir/src/Plugin_Scan.C.o" "CMakeFiles/nmxt-plugin-scan.dir/src/ladspa/LADSPAInfo.C.o" "CMakeFiles/nmxt-plugin-scan.dir/src/clap/Clap_Discovery.C.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst2/Vst2_Discovery.C.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/Vst3_Discovery.C.o" "CMakeFiles/nmxt-plugin-scan.dir/__/nonlib/debug.C.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/base/source/baseiids.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/base/source/fobject.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/base/thread/source/flock.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/public.sdk/source/common/commoniids.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/public.sdk/source/vst/vstinitiids.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/pluginterfaces/base/coreiids.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/vst3/vst3sdk/pluginterfaces/base/funknown.cpp.o" "CMakeFiles/nmxt-plugin-scan.dir/src/nmxt-plugin-scan.C.o" -o nmxt-plugin-scan  /usr/lib/x86_64-linux-gnu/libfltk_images.so /usr/lib/x86_64-linux-gnu/libfltk_forms.so /usr/lib/x86_64-linux-gnu/libfltk.so /usr/lib/x86_64-linux-gnu/libSM.so /usr/lib/x86_64-linux-gnu/libICE.so /usr/lib/x86_64-linux-gnu/libX11.so /usr/lib/x86_64-linux-gnu/libXext.so -lm -llilv-0 -lsuil-0 -lpthread -llrdf -ldl
```

Then "make" ran fine.
Any idea what's wrong?

**Comment by Stazed** [comment]
Just pushed a fix to the wip branch. There was a missing dl link but it did not seem to cause a problem with multiple other distributions I tested. Curious, what distribution, g++ version, and cmake version are you using?
Regardless, it should build now...

**Comment by suedwestlicht** [comment]
My system is a Xubuntu 16.04 pimped with repos from https://launchpad.net/~savoury1 so I can compile recent software.

I fucked it up because of some conflicts with manually compiled libs in /usr/local/lib and libs from the repos. I restored the system to the point before I installed the LV2 package from the repos that was too old for non-mixer-xt and tried to fix it.

Then I didn't have the problem with the missing -ldl

So I think it cannot hurt to add -ldl although it seems there isn't a problem on most systems when it's missing.
